src/components/pill-selector-dropdown/PillSelectorDropdown.js

```diff
@@ -177,21 +176,20 @@ class PillSelectorDropdown extends React.Component<Props, State> {
     };
 
     handleEnter = (event: SyntheticEvent<>) => {
-        const { isInCompositionMode } = this.state;
+        const { isInCompositionMode, inputValue } = this.state;
         if (!isInCompositionMode) {
             event.preventDefault();
-            this.addPillsFromInput();
+            this.addPillsFromInput(inputValue);
         }
     };
 
-    handlePaste = () => {
-        /**
-         * NOTE (ishay): setTimeout is necessary because
-         * otherwise addPillsFromInput gets triggered as soon
-         * as the user "paste's", but before the inputValue
-         * is actually updated.
-         */
-        setTimeout(this.addPillsFromInput, 0);
+    handlePaste = (event: SyntheticClipboardEvent<HTMLInputElement>) => {
+        event.nativeEvent.preventDefault();
+
+        const inputValue: string = event.clipboardData.getData('Text');
```

Comment:
Does this work in all browsers? Should it be `clipboardData.getData('text')` or `clipboardData.getData('text/plain')`, per [MDN](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/getData)?

Reply:
React appears to normalize the parts of the event specific to the Clipboard API. There used to be more details about the `.types` documented, but they took those off. I'll double-check if a non-legacy MIME type works here (e.g., `text/plain`)

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/types

Reply:
`Text` and `text` work in all browsers, but IE11 doesn't support the full MIME types. Will lower-case this